### PR TITLE
BREAKING: Simplify Internal Code, Use Generics, Change API, Add Client

### PR DESCRIPTION
**Pull request by ebebbington**
Take at simplifying wocket as much as possible to provide a clean api to uses and us, stripping back anything that isn't necessary

### CODE REVIEW

File: poop.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  console.log('got channel msg')
+  const { username, sender, id } = event.detail
+  
+  // Example sending to all clients, specifying which channel it is for
+  server.to("chat-message", {
+    message: "yep, the server got your msg :)"
+  })
+
+  // example sending to a specific client TODO :: Pass socket id as first param?
+  server.to("chat-message", {
```
Comment (crookse) [inline poop.ts:45]: is this example a "send a user a message that is connected to the chat-message channel" ?

File: poop.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    message: "this message is only for you *wink wink*"
+  }, {
+    id,
+    only: true
+  })
+
+  // example sending to all OTHER clients TODO :: broadcast?
+  server.to("chat-message", {
+    message: "You got this message but client with id of " + id + " shouldnt have"
```
Comment (crookse) [inline poop.ts:54]: yea this would be a broadcast

File: poop.ts (hunk 1 above)
Comment (ebebbington) [reply poop.ts:45]: yeah, just forgot to update the docblock, it'll send the message **only** to client with id of `id` in the `chat-message` channel

File: poop.ts (hunk 2 above)
Comment (ebebbington) [reply poop.ts:51]: https://github.com/drashland/wocket/pull/124/commits/13dfc572c46016def7f1c0cd46f085bf06489c2e

File: poop.ts (hunk 1 above)
Comment (ebebbington) [reply poop.ts:45]: https://github.com/drashland/wocket/pull/124/commits/66659eb7952d47fce56803033bb2c4b980c21f47

**Comment by ebebbington** [comment]
need time to check if this is finished, but its at a very solid state rn, not much will be added or changed

**Review by crookse — APPROVED** [review]
dang. LGTM.

**Pull request merged**